Applications/TextEditor/TextEditor.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <LibGUI/GApplication.h>
+#include <LibGUI/GTextEditor.h>
+#include <LibGUI/GWindow.h>
+
+class TextEditorWidget final : public GWidget {
+public:
+    TextEditorWidget(GApplication* app, GWindow* window, const int& argc, char** argv);
+    virtual ~TextEditorWidget() override;
+    void open_sesame(GWindow* window, GTextEditor* editor, const String& path);
```

Comment:
This shouldn't need the `GWindow` either (just use `window()`).
Also we should store the `GTextEditor` in a member variable instead of it being a temporary stack variable in the constructor:

```cpp
private:
    GTextEditor* m_editor { nullptr };
```

That way we can access the `GTextEditor` later on, long after the constructor has finished :)